app/src/test/java/org/mozilla/fenix/components/toolbar/BrowserInteractorTest.kt

```diff
@@ -18,59 +19,62 @@ import org.mozilla.fenix.quickactionsheet.QuickActionSheetController
 
 class BrowserInteractorTest {
 
+    val metrics: MetricController = mockk()
```

Comment:
In general, it's not a good idea to share state between tests. This way the tests can easily start depending on execution order e.g. one test can alter the shared state in way that makes another test fail.

A way to refactor this and avoid the side-effects is to use `@Before`:
```kotlin
    lateinit var metrics: MetricController
    lateinit var context: Context
    lateinit var browserStore: BrowserFragmentStore
    lateinit var browserToolbarController: BrowserToolbarController
    lateinit var quickActionSheetController: QuickActionSheetController
    lateinit var readerModeController: ReaderModeController
    lateinit var session: Session
    lateinit var interactor: BrowserInteractor

    @Before
    fun setup() {
        metrics = mockk()
        context = mockk()
        browserStore = mockk(relaxed = true)
        browserToolbarController = mockk(relaxed = true)
        quickActionSheetController = mockk(relaxed = true)
        readerModeController = mockk(relaxed = true)
        session = mockk()
        interactor = BrowserInteractor(
            context,
            browserStore,
            browserToolbarController,
            quickActionSheetController,
            readerModeController,
            session
        )
        every { context.metrics } returns metrics
        every { context.components.core.sessionManager.selectedSession } returns session
    }
```

This way every test starts with a fresh state and you still address the duplication.

Reply:
Good catch here! 😄